src/app/components/todos/todos.component.ts

```diff
@@ -20,14 +20,14 @@ export class TodosComponent implements OnInit {
   }
 
   deleteTodo(todo:Todo){
-    this.todos = this.todos.filter(t=>t.id!==todo.id);
-    this.todoService.deleteTodo(todo).subscribe();
+    this.todoService.addTodo(todo).subscribe(todo => {
```

Comment:
Consider handling potential errors from the `addTodo` observable by adding a second callback function to the `subscribe` method for error handling.